<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Supported differential expression methods • iSEEde</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Supported differential expression methods">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">iSEEde</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.3.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/iSEEde.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/methods.html">Supported differential expression methods</a></li>
    <li><a class="dropdown-item" href="../articles/annotations.html">Using annotations to facilitate interactive exploration</a></li>
    <li><a class="dropdown-item" href="../articles/rounding.html">Rounding numeric values</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/iSEE/iSEEde/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Supported differential expression methods</h1>
                        <h4 data-toc-skip class="author">Kevin
Rue-Albrecht</h4>
            <address class="author_afil">
      University of
Oxford<br><a class="author_email" href="mailto:#"></a><a href="mailto:kevin.rue-albrecht@imm.ox.ac.uk" class="email">kevin.rue-albrecht@imm.ox.ac.uk</a>
      </address>
                  
            <h4 data-toc-skip class="date">16 October 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/iSEE/iSEEde/blob/devel/vignettes/methods.Rmd" class="external-link"><code>vignettes/methods.Rmd</code></a></small>
      <div class="d-none name"><code>methods.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="implementation">Implementation<a class="anchor" aria-label="anchor" href="#implementation"></a>
</h2>
<div class="section level3">
<h3 id="user-facing-storage-and-access">User-facing storage and access<a class="anchor" aria-label="anchor" href="#user-facing-storage-and-access"></a>
</h3>
<p>Differential expression results are generally reported as tables of
statistics, including (log<sub>2</sub>) fold-change, p-value, average
expression, etc.</p>
<p>Those statistics being reported for individual features (e.g.,
genes), the <code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData()</a></code> component of
<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">SummarizedExperiment()</a></code> objects provides a natural home for
that information. Specifically, <em><a href="https://bioconductor.org/packages/3.20/iSEEde" class="external-link">iSEEde</a></em>
stores and retrieves differential expression results in
<code>rowData(se)[["iSEEde"]]</code>. However, the
<code><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData()</a></code> function should not be used to access or edit
that information. Instead, the functions
<code><a href="../reference/utils-SummarizedExperiment.html">embedContrastResults()</a></code> and <code><a href="../reference/contrastResults.html">contrastResults()</a></code>,
should be used to store and retrieve contrast results, respectively, as
those functions ensure that feature names are kept synchronised with the
enclosing <code>SummarizedExperiment</code> object.</p>
<p>Moreover, the function <code><a href="../reference/contrastResults.html">contrastResultsNames()</a></code> can be
used to retrieve the names of contrast available in a given
<code>SummarizedExperiment</code> object.</p>
</div>
<div class="section level3">
<h3 id="additional-considerations">Additional considerations<a class="anchor" aria-label="anchor" href="#additional-considerations"></a>
</h3>
<p>The first challenge arises when differential expression statistics
are computed only for a subset of features. In that case,
<code><a href="../reference/utils-SummarizedExperiment.html">embedContrastResults()</a></code> populates missing information with
<code>NA</code> values.</p>
<p>The second challenge arises from the different names of columns used
by individual differential expression methods to store differential
expression common statistics. To address this, <em><a href="https://bioconductor.org/packages/3.20/iSEEde" class="external-link">iSEEde</a></em>
provides S4 classes creating a common interface to supported
differential expression methods. Each class of differential expression
results implements the following methods:</p>
<ul>
<li>
<code><a href="../reference/de-generics.html">pValue()</a></code> returns the vector of raw p-values.</li>
<li>
<code><a href="../reference/de-generics.html">log2FoldChange()</a></code> returns the vector of log2-fold-change
values.</li>
<li>
<code><a href="../reference/de-generics.html">averageLog2()</a></code> returns the vector of average
log2-expression values.</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="example-data">Example data<a class="anchor" aria-label="anchor" href="#example-data"></a>
</h2>
<p>In this example, we use the <code><a href="https://rdrr.io/pkg/airway/man/airway.html" class="external-link">?airway</a></code> data set.</p>
<p>We briefly adjust the reference level of the treatment factor to the
untreated condition.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"airway"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"airway"</span><span class="op">)</span></span>
<span><span class="va">airway</span><span class="op">$</span><span class="va">dex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html" class="external-link">relevel</a></span><span class="op">(</span><span class="va">airway</span><span class="op">$</span><span class="va">dex</span>, <span class="st">"untrt"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="supported-methods">Supported methods<a class="anchor" aria-label="anchor" href="#supported-methods"></a>
</h2>
<div class="section level3">
<h3 id="limma">Limma<a class="anchor" aria-label="anchor" href="#limma"></a>
</h3>
<p>We first use <code><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html" class="external-link">edgeR::filterByExpr()</a></code> to remove genes whose
counts are too low to support a rigorous differential expression
analysis. Then we run a standard Limma-Voom analysis using
<code><a href="https://rdrr.io/pkg/edgeR/man/voomLmFit.html" class="external-link">edgeR::voomLmFit()</a></code>, <code><a href="https://rdrr.io/pkg/limma/man/makeContrasts.html" class="external-link">limma::makeContrasts()</a></code>,
and <code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">limma::eBayes()</a></code>. (Alternatively, we could have used
<code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">limma::treat()</a></code> instead of
<code><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">limma::eBayes()</a></code>.)</p>
<p>The linear model includes the <code>dex</code> and <code>cell</code>
covariates, indicating the treatment conditions and cell types,
respectively. Here, we are interested in differences between treatments,
adjusted by cell type, and define this comparison as the
<code>dextrt - dexuntrt</code> contrast.</p>
<p>The final differential expression results are fetched using
<code><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">limma::topTable()</a></code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioinf.wehi.edu.au/edgeR/" class="external-link">"edgeR"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">dex</span> <span class="op">+</span> <span class="va">cell</span>, data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html" class="external-link">filterByExpr</a></span><span class="op">(</span><span class="va">airway</span>, <span class="va">design</span><span class="op">)</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/voomLmFit.html" class="external-link">voomLmFit</a></span><span class="op">(</span><span class="va">airway</span><span class="op">[</span><span class="va">keep</span>, <span class="op">]</span>, <span class="va">design</span>, plot <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">contr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/makeContrasts.html" class="external-link">makeContrasts</a></span><span class="op">(</span><span class="st">"dextrt - dexuntrt"</span>, levels <span class="op">=</span> <span class="va">design</span><span class="op">)</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/contrasts.fit.html" class="external-link">contrasts.fit</a></span><span class="op">(</span><span class="va">fit</span>, <span class="va">contr</span><span class="op">)</span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit</span><span class="op">)</span></span>
<span><span class="va">res_limma</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">fit</span>, sort.by <span class="op">=</span> <span class="st">"P"</span>, n <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res_limma</span><span class="op">)</span></span>
<span><span class="co">#&gt;                         gene_id gene_name entrezid   gene_biotype gene_seq_start gene_seq_end</span></span>
<span><span class="co">#&gt; ENSG00000165995 ENSG00000165995    CACNB2       NA protein_coding       18429606     18830798</span></span>
<span><span class="co">#&gt; ENSG00000120129 ENSG00000120129     DUSP1       NA protein_coding      172195093    172198198</span></span>
<span><span class="co">#&gt; ENSG00000146250 ENSG00000146250    PRSS35       NA protein_coding       84222194     84235423</span></span>
<span><span class="co">#&gt; ENSG00000189221 ENSG00000189221      MAOA       NA protein_coding       43515467     43606068</span></span>
<span><span class="co">#&gt; ENSG00000157214 ENSG00000157214    STEAP2       NA protein_coding       89796904     89867451</span></span>
<span><span class="co">#&gt; ENSG00000152583 ENSG00000152583   SPARCL1       NA protein_coding       88394487     88452213</span></span>
<span><span class="co">#&gt;                 seq_name seq_strand seq_coord_system  symbol     logFC  AveExpr         t</span></span>
<span><span class="co">#&gt; ENSG00000165995       10          1               NA  CACNB2  3.205606 3.682244  37.68303</span></span>
<span><span class="co">#&gt; ENSG00000120129        5         -1               NA   DUSP1  2.864778 6.644455  28.50569</span></span>
<span><span class="co">#&gt; ENSG00000146250        6          1               NA  PRSS35 -2.828184 3.224885 -28.10830</span></span>
<span><span class="co">#&gt; ENSG00000189221        X          1               NA    MAOA  3.256085 5.950559  27.66135</span></span>
<span><span class="co">#&gt; ENSG00000157214        7          1               NA  STEAP2  1.894559 6.790009  27.40834</span></span>
<span><span class="co">#&gt; ENSG00000152583        4         -1               NA SPARCL1  4.489009 4.166904  27.34820</span></span>
<span><span class="co">#&gt;                      P.Value    adj.P.Val        B</span></span>
<span><span class="co">#&gt; ENSG00000165995 1.115938e-10 1.881472e-06 14.45518</span></span>
<span><span class="co">#&gt; ENSG00000120129 1.148539e-09 4.309046e-06 13.01074</span></span>
<span><span class="co">#&gt; ENSG00000146250 1.291089e-09 4.309046e-06 12.44376</span></span>
<span><span class="co">#&gt; ENSG00000189221 1.475547e-09 4.309046e-06 12.75540</span></span>
<span><span class="co">#&gt; ENSG00000157214 1.592916e-09 4.309046e-06 12.71210</span></span>
<span><span class="co">#&gt; ENSG00000152583 1.622327e-09 4.309046e-06 12.33402</span></span></code></pre></div>
<p>Then, we embed this set of differential expression results in the
<code>airway</code> object using the <code><a href="../reference/utils-SummarizedExperiment.html">embedContrastResults()</a></code>
method.</p>
<p>Because the output of <code><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">limma::topTable()</a></code> is a standard
<code>data.frame</code>, the <code>class=</code> argument must be used
to manually identify the method that produced those results.</p>
<p>Supported classes are listed in the object
<code><a href="../reference/utils-SummarizedExperiment.html">iSEEde::embedContrastResultsMethods</a></code>, i.e.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/iSEE/iSEEde" class="external-link">iSEEde</a></span><span class="op">)</span></span>
<span><span class="va">embedContrastResultsMethods</span></span>
<span><span class="co">#&gt;              limma </span></span>
<span><span class="co">#&gt; "iSEELimmaResults"</span></span></code></pre></div>
<p>This manual method is preferable to any automated heuristic
(e.g. using the column names of the <code>data.frame</code> for
identifying it as a <code><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">limma::topTable()</a></code> output).</p>
<p>The results embedded in the <code>airway</code> object can be
accessed using the <code><a href="../reference/contrastResults.html">contrastResults()</a></code> function.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">airway</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/utils-SummarizedExperiment.html">embedContrastResults</a></span><span class="op">(</span><span class="va">res_limma</span>, <span class="va">airway</span>, name <span class="op">=</span> <span class="st">"Limma-Voom"</span>, class <span class="op">=</span> <span class="st">"limma"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/contrastResults.html">contrastResults</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 63677 rows and 1 column</span></span>
<span><span class="co">#&gt;                         Limma-Voom</span></span>
<span><span class="co">#&gt;                 &lt;iSEELimmaResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000003 &lt;iSEELimmaResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000005 &lt;iSEELimmaResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000419 &lt;iSEELimmaResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000457 &lt;iSEELimmaResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000460 &lt;iSEELimmaResults&gt;</span></span>
<span><span class="co">#&gt; ...                            ...</span></span>
<span><span class="co">#&gt; ENSG00000273489 &lt;iSEELimmaResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000273490 &lt;iSEELimmaResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000273491 &lt;iSEELimmaResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000273492 &lt;iSEELimmaResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000273493 &lt;iSEELimmaResults&gt;</span></span>
<span><span class="fu"><a href="../reference/contrastResults.html">contrastResults</a></span><span class="op">(</span><span class="va">airway</span>, <span class="st">"Limma-Voom"</span><span class="op">)</span></span>
<span><span class="co">#&gt; iSEELimmaResults with 63677 rows and 16 columns</span></span>
<span><span class="co">#&gt;                         gene_id   gene_name  entrezid   gene_biotype gene_seq_start gene_seq_end</span></span>
<span><span class="co">#&gt;                     &lt;character&gt; &lt;character&gt; &lt;integer&gt;    &lt;character&gt;      &lt;integer&gt;    &lt;integer&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000003 ENSG00000000003      TSPAN6        NA protein_coding       99883667     99894988</span></span>
<span><span class="co">#&gt; ENSG00000000005              NA          NA        NA             NA             NA           NA</span></span>
<span><span class="co">#&gt; ENSG00000000419 ENSG00000000419        DPM1        NA protein_coding       49551404     49575092</span></span>
<span><span class="co">#&gt; ENSG00000000457 ENSG00000000457       SCYL3        NA protein_coding      169818772    169863408</span></span>
<span><span class="co">#&gt; ENSG00000000460 ENSG00000000460    C1orf112        NA protein_coding      169631245    169823221</span></span>
<span><span class="co">#&gt; ...                         ...         ...       ...            ...            ...          ...</span></span>
<span><span class="co">#&gt; ENSG00000273489              NA          NA        NA             NA             NA           NA</span></span>
<span><span class="co">#&gt; ENSG00000273490              NA          NA        NA             NA             NA           NA</span></span>
<span><span class="co">#&gt; ENSG00000273491              NA          NA        NA             NA             NA           NA</span></span>
<span><span class="co">#&gt; ENSG00000273492              NA          NA        NA             NA             NA           NA</span></span>
<span><span class="co">#&gt; ENSG00000273493              NA          NA        NA             NA             NA           NA</span></span>
<span><span class="co">#&gt;                    seq_name seq_strand seq_coord_system      symbol     logFC   AveExpr         t</span></span>
<span><span class="co">#&gt;                 &lt;character&gt;  &lt;integer&gt;        &lt;integer&gt; &lt;character&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000003           X         -1               NA      TSPAN6 -0.464216   5.02559 -6.589673</span></span>
<span><span class="co">#&gt; ENSG00000000005          NA         NA               NA          NA        NA        NA        NA</span></span>
<span><span class="co">#&gt; ENSG00000000419          20         -1               NA        DPM1  0.125077   4.60191  1.632802</span></span>
<span><span class="co">#&gt; ENSG00000000457           1         -1               NA       SCYL3 -0.042107   3.47269 -0.438929</span></span>
<span><span class="co">#&gt; ENSG00000000460           1          1               NA    C1orf112 -0.228517   1.40857 -0.977961</span></span>
<span><span class="co">#&gt; ...                     ...        ...              ...         ...       ...       ...       ...</span></span>
<span><span class="co">#&gt; ENSG00000273489          NA         NA               NA          NA        NA        NA        NA</span></span>
<span><span class="co">#&gt; ENSG00000273490          NA         NA               NA          NA        NA        NA        NA</span></span>
<span><span class="co">#&gt; ENSG00000273491          NA         NA               NA          NA        NA        NA        NA</span></span>
<span><span class="co">#&gt; ENSG00000273492          NA         NA               NA          NA        NA        NA        NA</span></span>
<span><span class="co">#&gt; ENSG00000273493          NA         NA               NA          NA        NA        NA        NA</span></span>
<span><span class="co">#&gt;                     P.Value  adj.P.Val         B</span></span>
<span><span class="co">#&gt;                   &lt;numeric&gt;  &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000003 0.000136603 0.00171857  0.899306</span></span>
<span><span class="co">#&gt; ENSG00000000005          NA         NA        NA</span></span>
<span><span class="co">#&gt; ENSG00000000419 0.139297605 0.25279527 -6.274792</span></span>
<span><span class="co">#&gt; ENSG00000000457 0.671768770 0.77787936 -7.230811</span></span>
<span><span class="co">#&gt; ENSG00000000460 0.355374953 0.49644090 -6.208897</span></span>
<span><span class="co">#&gt; ...                     ...        ...       ...</span></span>
<span><span class="co">#&gt; ENSG00000273489          NA         NA        NA</span></span>
<span><span class="co">#&gt; ENSG00000273490          NA         NA        NA</span></span>
<span><span class="co">#&gt; ENSG00000273491          NA         NA        NA</span></span>
<span><span class="co">#&gt; ENSG00000273492          NA         NA        NA</span></span>
<span><span class="co">#&gt; ENSG00000273493          NA         NA        NA</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="deseq2">DESeq2<a class="anchor" aria-label="anchor" href="#deseq2"></a>
</h3>
<p>First, we use <code><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet()</a></code> to construct a
<code>DESeqDataSet</code> object for the analysis. Then we run a
standard <em><a href="https://bioconductor.org/packages/3.20/DESeq2" class="external-link">DESeq2</a></em>
analysis using <code><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq()</a></code>.</p>
<p>The differential expression results are fetched using
<code><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results()</a></code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/thelovelab/DESeq2" class="external-link">"DESeq2"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></span><span class="op">(</span><span class="va">airway</span>, <span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">dex</span> <span class="op">+</span> <span class="va">cell</span><span class="op">)</span></span>
<span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span>
<span><span class="va">res_deseq2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds</span>, contrast <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"dextrt"</span>, <span class="st">"dexuntrt"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res_deseq2</span><span class="op">)</span></span>
<span><span class="co">#&gt; log2 fold change (MLE): dextrt vs dexuntrt </span></span>
<span><span class="co">#&gt; Wald test p-value: dextrt vs dexuntrt </span></span>
<span><span class="co">#&gt; DataFrame with 6 rows and 6 columns</span></span>
<span><span class="co">#&gt;                   baseMean log2FoldChange     lfcSE      stat      pvalue       padj</span></span>
<span><span class="co">#&gt;                  &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;  &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000003 708.602170     -0.3812539  0.100654 -3.787752 0.000152016 0.00128292</span></span>
<span><span class="co">#&gt; ENSG00000000005   0.000000             NA        NA        NA          NA         NA</span></span>
<span><span class="co">#&gt; ENSG00000000419 520.297901      0.2068127  0.112219  1.842944 0.065337213 0.19646961</span></span>
<span><span class="co">#&gt; ENSG00000000457 237.163037      0.0379205  0.143445  0.264356 0.791505314 0.91141884</span></span>
<span><span class="co">#&gt; ENSG00000000460  57.932633     -0.0881679  0.287142 -0.307054 0.758802543 0.89500551</span></span>
<span><span class="co">#&gt; ENSG00000000938   0.318098     -1.3782416  3.499906 -0.393794 0.693733216         NA</span></span></code></pre></div>
<p>Then, we embed this set of differential expression results in the
<code>airway</code> object using the <code><a href="../reference/utils-SummarizedExperiment.html">embedContrastResults()</a></code>
method.</p>
<p>In this instance, the <em><a href="https://bioconductor.org/packages/3.20/DESeq2" class="external-link">DESeq2</a></em>
results are stored in a recognisable <code><a href="https://rdrr.io/pkg/DESeq2/man/DESeqResults.html" class="external-link">?DESeqResults</a></code> object,
which can be given <em>as is</em> directly to the
<code><a href="../reference/utils-SummarizedExperiment.html">embedContrastResults()</a></code> method.</p>
<p>The function will automatically pass the results object to the
<code><a href="../reference/iSEEDESeq2Results-class.html">iSEEDESeq2Results()</a></code> constructor, to identify it as
such.</p>
<p>The results embedded in the airway object can be accessed using the
<code><a href="../reference/contrastResults.html">contrastResults()</a></code> function.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">airway</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/utils-SummarizedExperiment.html">embedContrastResults</a></span><span class="op">(</span><span class="va">res_deseq2</span>, <span class="va">airway</span>, name <span class="op">=</span> <span class="st">"DESeq2"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/contrastResults.html">contrastResults</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 63677 rows and 2 columns</span></span>
<span><span class="co">#&gt;                         Limma-Voom              DESeq2</span></span>
<span><span class="co">#&gt;                 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000003 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000005 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000419 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000457 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000460 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt;</span></span>
<span><span class="co">#&gt; ...                            ...                 ...</span></span>
<span><span class="co">#&gt; ENSG00000273489 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt;</span></span>
<span><span class="co">#&gt; ENSG00000273490 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt;</span></span>
<span><span class="co">#&gt; ENSG00000273491 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt;</span></span>
<span><span class="co">#&gt; ENSG00000273492 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt;</span></span>
<span><span class="co">#&gt; ENSG00000273493 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt;</span></span>
<span><span class="fu"><a href="../reference/contrastResults.html">contrastResults</a></span><span class="op">(</span><span class="va">airway</span>, <span class="st">"DESeq2"</span><span class="op">)</span></span>
<span><span class="co">#&gt; iSEEDESeq2Results with 63677 rows and 6 columns</span></span>
<span><span class="co">#&gt;                  baseMean log2FoldChange     lfcSE      stat      pvalue       padj</span></span>
<span><span class="co">#&gt;                 &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;  &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000003  708.6022     -0.3812539  0.100654 -3.787752 0.000152016 0.00128292</span></span>
<span><span class="co">#&gt; ENSG00000000005    0.0000             NA        NA        NA          NA         NA</span></span>
<span><span class="co">#&gt; ENSG00000000419  520.2979      0.2068127  0.112219  1.842944 0.065337213 0.19646961</span></span>
<span><span class="co">#&gt; ENSG00000000457  237.1630      0.0379205  0.143445  0.264356 0.791505314 0.91141884</span></span>
<span><span class="co">#&gt; ENSG00000000460   57.9326     -0.0881679  0.287142 -0.307054 0.758802543 0.89500551</span></span>
<span><span class="co">#&gt; ...                   ...            ...       ...       ...         ...        ...</span></span>
<span><span class="co">#&gt; ENSG00000273489  0.275899       1.483744   3.51398  0.422240    0.672850         NA</span></span>
<span><span class="co">#&gt; ENSG00000273490  0.000000             NA        NA        NA          NA         NA</span></span>
<span><span class="co">#&gt; ENSG00000273491  0.000000             NA        NA        NA          NA         NA</span></span>
<span><span class="co">#&gt; ENSG00000273492  0.105978      -0.463688   3.52312 -0.131613    0.895290         NA</span></span>
<span><span class="co">#&gt; ENSG00000273493  0.106142      -0.521372   3.53142 -0.147638    0.882628         NA</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="edger">edgeR<a class="anchor" aria-label="anchor" href="#edger"></a>
</h3>
<p>We run a standard <em><a href="https://bioconductor.org/packages/3.20/edgeR" class="external-link">edgeR</a></em>
analysis using <code><a href="https://rdrr.io/pkg/edgeR/man/glmfit.html" class="external-link">glmFit()</a></code> and <code><a href="https://rdrr.io/pkg/edgeR/man/glmLRT.html" class="external-link">glmLRT()</a></code>.</p>
<p>The differential expression results are fetched using
<code><a href="https://rdrr.io/pkg/edgeR/man/topTags.html" class="external-link">topTags()</a></code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://bioinf.wehi.edu.au/edgeR/" class="external-link">"edgeR"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="fl">0</span> <span class="op">+</span> <span class="va">dex</span> <span class="op">+</span> <span class="va">cell</span>, data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmfit.html" class="external-link">glmFit</a></span><span class="op">(</span><span class="va">airway</span>, <span class="va">design</span>, dispersion <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span></span>
<span><span class="va">lrt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/glmLRT.html" class="external-link">glmLRT</a></span><span class="op">(</span><span class="va">fit</span>, contrast <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span>, <span class="fl">1</span>, <span class="fl">0</span>, <span class="fl">0</span>, <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">res_edger</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/topTags.html" class="external-link">topTags</a></span><span class="op">(</span><span class="va">lrt</span>, n <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">res_edger</span><span class="op">)</span></span>
<span><span class="co">#&gt; Coefficient:  -1*dexuntrt 1*dextrt </span></span>
<span><span class="co">#&gt;                         gene_id     gene_name entrezid         gene_biotype gene_seq_start</span></span>
<span><span class="co">#&gt; ENSG00000109906 ENSG00000109906        ZBTB16       NA       protein_coding      113930315</span></span>
<span><span class="co">#&gt; ENSG00000179593 ENSG00000179593       ALOX15B       NA       protein_coding        7942335</span></span>
<span><span class="co">#&gt; ENSG00000127954 ENSG00000127954        STEAP4       NA       protein_coding       87905744</span></span>
<span><span class="co">#&gt; ENSG00000152583 ENSG00000152583       SPARCL1       NA       protein_coding       88394487</span></span>
<span><span class="co">#&gt; ENSG00000250978 ENSG00000250978 RP11-357D18.1       NA processed_transcript       66759637</span></span>
<span><span class="co">#&gt; ENSG00000163884 ENSG00000163884         KLF15       NA       protein_coding      126061478</span></span>
<span><span class="co">#&gt;                 gene_seq_end seq_name seq_strand seq_coord_system        symbol</span></span>
<span><span class="co">#&gt; ENSG00000109906    114121398       11          1               NA        ZBTB16</span></span>
<span><span class="co">#&gt; ENSG00000179593      7952452       17          1               NA       ALOX15B</span></span>
<span><span class="co">#&gt; ENSG00000127954     87936206        7         -1               NA        STEAP4</span></span>
<span><span class="co">#&gt; ENSG00000152583     88452213        4         -1               NA       SPARCL1</span></span>
<span><span class="co">#&gt; ENSG00000250978     66771420        5         -1               NA RP11-357D18.1</span></span>
<span><span class="co">#&gt; ENSG00000163884    126076285        3         -1               NA         KLF15</span></span>
<span><span class="co">#&gt;                 iSEEde.Limma.Voom.gene_id iSEEde.Limma.Voom.gene_name iSEEde.Limma.Voom.entrezid</span></span>
<span><span class="co">#&gt; ENSG00000109906           ENSG00000109906                      ZBTB16                         NA</span></span>
<span><span class="co">#&gt; ENSG00000179593           ENSG00000179593                     ALOX15B                         NA</span></span>
<span><span class="co">#&gt; ENSG00000127954           ENSG00000127954                      STEAP4                         NA</span></span>
<span><span class="co">#&gt; ENSG00000152583           ENSG00000152583                     SPARCL1                         NA</span></span>
<span><span class="co">#&gt; ENSG00000250978           ENSG00000250978               RP11-357D18.1                         NA</span></span>
<span><span class="co">#&gt; ENSG00000163884           ENSG00000163884                       KLF15                         NA</span></span>
<span><span class="co">#&gt;                 iSEEde.Limma.Voom.gene_biotype iSEEde.Limma.Voom.gene_seq_start</span></span>
<span><span class="co">#&gt; ENSG00000109906                 protein_coding                        113930315</span></span>
<span><span class="co">#&gt; ENSG00000179593                 protein_coding                          7942335</span></span>
<span><span class="co">#&gt; ENSG00000127954                 protein_coding                         87905744</span></span>
<span><span class="co">#&gt; ENSG00000152583                 protein_coding                         88394487</span></span>
<span><span class="co">#&gt; ENSG00000250978           processed_transcript                         66759637</span></span>
<span><span class="co">#&gt; ENSG00000163884                 protein_coding                        126061478</span></span>
<span><span class="co">#&gt;                 iSEEde.Limma.Voom.gene_seq_end iSEEde.Limma.Voom.seq_name</span></span>
<span><span class="co">#&gt; ENSG00000109906                      114121398                         11</span></span>
<span><span class="co">#&gt; ENSG00000179593                        7952452                         17</span></span>
<span><span class="co">#&gt; ENSG00000127954                       87936206                          7</span></span>
<span><span class="co">#&gt; ENSG00000152583                       88452213                          4</span></span>
<span><span class="co">#&gt; ENSG00000250978                       66771420                          5</span></span>
<span><span class="co">#&gt; ENSG00000163884                      126076285                          3</span></span>
<span><span class="co">#&gt;                 iSEEde.Limma.Voom.seq_strand iSEEde.Limma.Voom.seq_coord_system</span></span>
<span><span class="co">#&gt; ENSG00000109906                            1                                 NA</span></span>
<span><span class="co">#&gt; ENSG00000179593                            1                                 NA</span></span>
<span><span class="co">#&gt; ENSG00000127954                           -1                                 NA</span></span>
<span><span class="co">#&gt; ENSG00000152583                           -1                                 NA</span></span>
<span><span class="co">#&gt; ENSG00000250978                           -1                                 NA</span></span>
<span><span class="co">#&gt; ENSG00000163884                           -1                                 NA</span></span>
<span><span class="co">#&gt;                 iSEEde.Limma.Voom.symbol iSEEde.Limma.Voom.logFC iSEEde.Limma.Voom.AveExpr</span></span>
<span><span class="co">#&gt; ENSG00000109906                   ZBTB16                7.069003                 1.3807218</span></span>
<span><span class="co">#&gt; ENSG00000179593                  ALOX15B                7.988755                -1.4798754</span></span>
<span><span class="co">#&gt; ENSG00000127954                   STEAP4                5.184088                 1.6193300</span></span>
<span><span class="co">#&gt; ENSG00000152583                  SPARCL1                4.489009                 4.1669039</span></span>
<span><span class="co">#&gt; ENSG00000250978            RP11-357D18.1                6.031081                -0.7521822</span></span>
<span><span class="co">#&gt; ENSG00000163884                    KLF15                4.433096                 3.2401058</span></span>
<span><span class="co">#&gt;                 iSEEde.Limma.Voom.t iSEEde.Limma.Voom.P.Value iSEEde.Limma.Voom.adj.P.Val</span></span>
<span><span class="co">#&gt; ENSG00000109906            23.94855              4.892822e-09                4.852528e-06</span></span>
<span><span class="co">#&gt; ENSG00000179593            21.82342              1.743170e-06                1.076551e-04</span></span>
<span><span class="co">#&gt; ENSG00000127954            20.01478              2.162145e-08                1.012605e-05</span></span>
<span><span class="co">#&gt; ENSG00000152583            27.34820              1.622327e-09                4.309046e-06</span></span>
<span><span class="co">#&gt; ENSG00000250978            16.32481              1.154612e-07                2.187276e-05</span></span>
<span><span class="co">#&gt; ENSG00000163884            22.89074              7.117995e-09                5.425717e-06</span></span>
<span><span class="co">#&gt;                 iSEEde.Limma.Voom.B iSEEde.DESeq2.baseMean iSEEde.DESeq2.log2FoldChange</span></span>
<span><span class="co">#&gt; ENSG00000109906            8.652823              385.07103                     7.352629</span></span>
<span><span class="co">#&gt; ENSG00000179593            2.986534               67.24305                     9.505983</span></span>
<span><span class="co">#&gt; ENSG00000127954            9.354944              286.38412                     5.207161</span></span>
<span><span class="co">#&gt; ENSG00000152583           12.334016              997.43977                     4.574919</span></span>
<span><span class="co">#&gt; ENSG00000250978            6.074174               56.31819                     6.327386</span></span>
<span><span class="co">#&gt; ENSG00000163884           10.885003              561.10717                     4.459129</span></span>
<span><span class="co">#&gt;                 iSEEde.DESeq2.lfcSE iSEEde.DESeq2.stat iSEEde.DESeq2.pvalue iSEEde.DESeq2.padj</span></span>
<span><span class="co">#&gt; ENSG00000109906           0.5363902          13.707612         9.141717e-43       2.256376e-40</span></span>
<span><span class="co">#&gt; ENSG00000179593           1.0545033           9.014654         1.974926e-19       1.252965e-17</span></span>
<span><span class="co">#&gt; ENSG00000127954           0.4930839          10.560396         4.547384e-26       5.057701e-24</span></span>
<span><span class="co">#&gt; ENSG00000152583           0.1840561          24.856114        2.220777e-136      4.001395e-132</span></span>
<span><span class="co">#&gt; ENSG00000250978           0.6777980           9.335209         1.007922e-20       7.206645e-19</span></span>
<span><span class="co">#&gt; ENSG00000163884           0.2348638          18.986019         2.225778e-80       2.506504e-77</span></span>
<span><span class="co">#&gt;                     logFC   logCPM       LR       PValue          FDR</span></span>
<span><span class="co">#&gt; ENSG00000109906  7.183385 4.132638 238.3947 8.805179e-54 5.606874e-49</span></span>
<span><span class="co">#&gt; ENSG00000179593 10.015847 1.627629 181.0331 2.883024e-41 9.179116e-37</span></span>
<span><span class="co">#&gt; ENSG00000127954  5.087069 3.672567 146.9725 7.957020e-34 1.688931e-29</span></span>
<span><span class="co">#&gt; ENSG00000152583  4.498698 5.510213 140.2205 2.382274e-32 3.792402e-28</span></span>
<span><span class="co">#&gt; ENSG00000250978  6.128131 1.377260 137.4681 9.526183e-32 1.213198e-27</span></span>
<span><span class="co">#&gt; ENSG00000163884  4.367962 4.681216 129.2203 6.069471e-30 6.441428e-26</span></span></code></pre></div>
<p>Then, we embed this set of differential expression results in the
<code>airway</code> object using the <code><a href="../reference/utils-SummarizedExperiment.html">embedContrastResults()</a></code>
method.</p>
<p>In this instance, the <em><a href="https://bioconductor.org/packages/3.20/edgeR" class="external-link">edgeR</a></em>
results are stored in a recognisable <code>?TopTags</code> object. As
such, the results object can be given <em>as is</em> directly to the
<code><a href="../reference/utils-SummarizedExperiment.html">embedContrastResults()</a></code> method. The function will
automatically pass the results object to the
<code><a href="../reference/iSEEedgeRResults-class.html">iSEEedgeRResults()</a></code> constructor, to identify it as such.</p>
<p>The results embedded in the airway object can be accessed using the
<code><a href="../reference/contrastResults.html">contrastResults()</a></code> function.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">airway</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/utils-SummarizedExperiment.html">embedContrastResults</a></span><span class="op">(</span><span class="va">res_edger</span>, <span class="va">airway</span>, name <span class="op">=</span> <span class="st">"edgeR"</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/contrastResults.html">contrastResults</a></span><span class="op">(</span><span class="va">airway</span><span class="op">)</span></span>
<span><span class="co">#&gt; DataFrame with 63677 rows and 3 columns</span></span>
<span><span class="co">#&gt;                         Limma-Voom              DESeq2              edgeR</span></span>
<span><span class="co">#&gt;                 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt; &lt;iSEEedgeRResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000003 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt; &lt;iSEEedgeRResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000005 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt; &lt;iSEEedgeRResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000419 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt; &lt;iSEEedgeRResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000457 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt; &lt;iSEEedgeRResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000460 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt; &lt;iSEEedgeRResults&gt;</span></span>
<span><span class="co">#&gt; ...                            ...                 ...                ...</span></span>
<span><span class="co">#&gt; ENSG00000273489 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt; &lt;iSEEedgeRResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000273490 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt; &lt;iSEEedgeRResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000273491 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt; &lt;iSEEedgeRResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000273492 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt; &lt;iSEEedgeRResults&gt;</span></span>
<span><span class="co">#&gt; ENSG00000273493 &lt;iSEELimmaResults&gt; &lt;iSEEDESeq2Results&gt; &lt;iSEEedgeRResults&gt;</span></span>
<span><span class="fu"><a href="../reference/contrastResults.html">contrastResults</a></span><span class="op">(</span><span class="va">airway</span>, <span class="st">"edgeR"</span><span class="op">)</span></span>
<span><span class="co">#&gt; iSEEedgeRResults with 63677 rows and 5 columns</span></span>
<span><span class="co">#&gt;                      logFC    logCPM        LR    PValue       FDR</span></span>
<span><span class="co">#&gt;                  &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;</span></span>
<span><span class="co">#&gt; ENSG00000000003 -0.4628153   5.05930  2.018481  0.155394         1</span></span>
<span><span class="co">#&gt; ENSG00000000005  0.0000000  -3.45546  0.000000  1.000000         1</span></span>
<span><span class="co">#&gt; ENSG00000000419  0.1247724   4.60783  0.146545  0.701860         1</span></span>
<span><span class="co">#&gt; ENSG00000000457 -0.0445216   3.48326  0.018241  0.892565         1</span></span>
<span><span class="co">#&gt; ENSG00000000460 -0.1618126   1.48518  0.210342  0.646500         1</span></span>
<span><span class="co">#&gt; ...                    ...       ...       ...       ...       ...</span></span>
<span><span class="co">#&gt; ENSG00000273489    2.48209  -3.28549   3.02143  0.082171         1</span></span>
<span><span class="co">#&gt; ENSG00000273490    0.00000  -3.45546   0.00000  1.000000         1</span></span>
<span><span class="co">#&gt; ENSG00000273491    0.00000  -3.45546   0.00000  1.000000         1</span></span>
<span><span class="co">#&gt; ENSG00000273492   -1.24012  -3.36894   0.91097  0.339857         1</span></span>
<span><span class="co">#&gt; ENSG00000273493   -1.75243  -3.36862   1.57193  0.209928         1</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="live-app">Live app<a class="anchor" aria-label="anchor" href="#live-app"></a>
</h2>
<p>In this example, we used the <em><a href="https://bioconductor.org/packages/3.20/iSEEde" class="external-link">iSEEde</a></em>
functions <code><a href="../reference/DETable-class.html">DETable()</a></code>, <code><a href="../reference/VolcanoPlot-class.html">VolcanoPlot()</a></code>, and
<code><a href="../reference/MAPlot-class.html">MAPlot()</a></code> to add panels that facilitate the visualisation of
differential expression results in an <em><a href="https://bioconductor.org/packages/3.20/iSEE" class="external-link">iSEE</a></em>
app.</p>
<p>Specifically, we add one set of panels for each differential
expression method used in this vignette (i.e., Limma-Voom, DESeq2,
edgeR).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/iSEE/iSEE" class="external-link">iSEE</a></span><span class="op">)</span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">airway</span>, initial <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="fu"><a href="../reference/DETable-class.html">DETable</a></span><span class="op">(</span>ContrastName<span class="op">=</span><span class="st">"Limma-Voom"</span>, HiddenColumns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"AveExpr"</span>, </span>
<span>    <span class="st">"t"</span>, <span class="st">"B"</span><span class="op">)</span>, PanelWidth <span class="op">=</span> <span class="fl">4L</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/VolcanoPlot-class.html">VolcanoPlot</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"Limma-Voom"</span>, PanelWidth <span class="op">=</span> <span class="fl">4L</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/MAPlot-class.html">MAPlot</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"Limma-Voom"</span>, PanelWidth <span class="op">=</span> <span class="fl">4L</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/DETable-class.html">DETable</a></span><span class="op">(</span>ContrastName<span class="op">=</span><span class="st">"DESeq2"</span>, HiddenColumns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"baseMean"</span>, </span>
<span>    <span class="st">"lfcSE"</span>, <span class="st">"stat"</span><span class="op">)</span>, PanelWidth <span class="op">=</span> <span class="fl">4L</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/VolcanoPlot-class.html">VolcanoPlot</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"DESeq2"</span>, PanelWidth <span class="op">=</span> <span class="fl">4L</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/MAPlot-class.html">MAPlot</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"DESeq2"</span>, PanelWidth <span class="op">=</span> <span class="fl">4L</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/DETable-class.html">DETable</a></span><span class="op">(</span>ContrastName<span class="op">=</span><span class="st">"edgeR"</span>, HiddenColumns <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"logCPM"</span>, </span>
<span>    <span class="st">"LR"</span><span class="op">)</span>, PanelWidth <span class="op">=</span> <span class="fl">4L</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/VolcanoPlot-class.html">VolcanoPlot</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"edgeR"</span>, PanelWidth <span class="op">=</span> <span class="fl">4L</span><span class="op">)</span>,</span>
<span>  <span class="fu"><a href="../reference/MAPlot-class.html">MAPlot</a></span><span class="op">(</span>ContrastName <span class="op">=</span> <span class="st">"edgeR"</span>, PanelWidth <span class="op">=</span> <span class="fl">4L</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="screenshots%2Fmethods_side_by_side.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="comparing-two-contrasts">Comparing two contrasts<a class="anchor" aria-label="anchor" href="#comparing-two-contrasts"></a>
</h2>
<p>The <code><a href="../reference/LogFCLogFCPlot-class.html">?LogFCLogFCPlot</a></code> class allows users to compare the
log<sub>2</sub> fold-change value of features between two differential
expression contrasts.</p>
<p>In this example, we add one <code><a href="../reference/LogFCLogFCPlot-class.html">?LogFCLogFCPlot</a></code> panel
comparing the same contrast using the Limma-Voom and DESeq2 methods,
alongside one <code><a href="../reference/VolcanoPlot-class.html">?VolcanoPlot</a></code> panel for each of those two
contrasts. Moreover, we pre-select an area of the
<code><a href="../reference/LogFCLogFCPlot-class.html">?LogFCLogFCPlot</a></code> and highlight the selected features in the
two <code><a href="../reference/VolcanoPlot-class.html">?VolcanoPlot</a></code> panels.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/iSEE/iSEE" class="external-link">iSEE</a></span><span class="op">)</span></span>
<span><span class="va">app</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/iSEE/man/iSEE.html" class="external-link">iSEE</a></span><span class="op">(</span><span class="va">airway</span>, initial <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>    <span class="fu"><a href="../reference/VolcanoPlot-class.html">VolcanoPlot</a></span><span class="op">(</span>ContrastName<span class="op">=</span><span class="st">"Limma-Voom"</span>,</span>
<span>        RowSelectionSource <span class="op">=</span> <span class="st">"LogFCLogFCPlot1"</span>, ColorBy <span class="op">=</span> <span class="st">"Row selection"</span>,</span>
<span>        PanelWidth <span class="op">=</span> <span class="fl">4L</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="../reference/LogFCLogFCPlot-class.html">LogFCLogFCPlot</a></span><span class="op">(</span>ContrastNameX<span class="op">=</span><span class="st">"Limma-Voom"</span>, ContrastNameY<span class="op">=</span><span class="st">"DESeq2"</span>,</span>
<span>        BrushData <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>        xmin <span class="op">=</span> <span class="fl">3.6</span>, xmax <span class="op">=</span> <span class="fl">8.2</span>, ymin <span class="op">=</span> <span class="fl">3.8</span>, ymax <span class="op">=</span> <span class="fl">9.8</span>,</span>
<span>        mapping <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"X"</span>, y <span class="op">=</span> <span class="st">"Y"</span><span class="op">)</span>,</span>
<span>        direction <span class="op">=</span> <span class="st">"xy"</span>, brushId <span class="op">=</span> <span class="st">"LogFCLogFCPlot1_Brush"</span>, </span>
<span>        outputId <span class="op">=</span> <span class="st">"LogFCLogFCPlot1"</span><span class="op">)</span>,</span>
<span>        PanelWidth <span class="op">=</span> <span class="fl">4L</span><span class="op">)</span>,</span>
<span>    <span class="fu"><a href="../reference/VolcanoPlot-class.html">VolcanoPlot</a></span><span class="op">(</span>ContrastName<span class="op">=</span><span class="st">"DESeq2"</span>,</span>
<span>        RowSelectionSource <span class="op">=</span> <span class="st">"LogFCLogFCPlot1"</span>, ColorBy <span class="op">=</span> <span class="st">"Row selection"</span>,</span>
<span>        PanelWidth <span class="op">=</span> <span class="fl">4L</span><span class="op">)</span></span>
<span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/interactive.html" class="external-link">interactive</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu">shiny</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/shiny/man/runApp.html" class="external-link">runApp</a></span><span class="op">(</span><span class="va">app</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p><img src="screenshots%2Flogfc_logfc_plot.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="reproducibility">Reproducibility<a class="anchor" aria-label="anchor" href="#reproducibility"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.20/iSEEde" class="external-link">iSEEde</a></em>
package (Rue-Albrecht, 2024) was made possible thanks to:</p>
<ul>
<li>R (R Core Team, 2024)</li>
<li>
<em><a href="https://bioconductor.org/packages/3.20/BiocStyle" class="external-link">BiocStyle</a></em>
(Oleś, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux, McPherson, Luraschi, Ushey, Atkins, Wickham,
Cheng, Chang, and Iannone, 2024)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=sessioninfo" class="external-link">sessioninfo</a></em>
(Wickham, Chang, Flight, Müller, and Hester, 2021)</li>
<li>
<em><a href="https://CRAN.R-project.org/package=testthat" class="external-link">testthat</a></em>
(Wickham, 2011)</li>
</ul>
<p>This package was developed using <em><a href="https://bioconductor.org/packages/3.20/biocthis" class="external-link">biocthis</a></em>.</p>
<p>Code for creating the vignette</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Create the vignette</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/rstudio/rmarkdown" class="external-link">"rmarkdown"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/system.time.html" class="external-link">system.time</a></span><span class="op">(</span><span class="fu"><a href="https://pkgs.rstudio.com/rmarkdown/reference/render.html" class="external-link">render</a></span><span class="op">(</span><span class="st">"methods.Rmd"</span>, <span class="st">"BiocStyle::html_document"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">## Extract the R code</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yihui.org/knitr/" class="external-link">"knitr"</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/knit.html" class="external-link">knit</a></span><span class="op">(</span><span class="st">"methods.Rmd"</span>, tangle <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>Date the vignette was generated.</p>
<pre><code><span><span class="co">#&gt; [1] "2024-10-16 08:36:36 UTC"</span></span></code></pre>
<p>Wallclock time spent generating the vignette.</p>
<pre><code><span><span class="co">#&gt; Time difference of 26.619 secs</span></span></code></pre>
<p><code>R</code> session information.</p>
<pre><code><span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Session info ───────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">setting </span> <span style="color: #555555; font-style: italic;">value</span></span></span>
<span><span class="co">#&gt;  version  R version 4.4.1 (2024-06-14)</span></span>
<span><span class="co">#&gt;  os       Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">#&gt;  system   x86_64, linux-gnu</span></span>
<span><span class="co">#&gt;  ui       X11</span></span>
<span><span class="co">#&gt;  language en</span></span>
<span><span class="co">#&gt;  collate  en_US.UTF-8</span></span>
<span><span class="co">#&gt;  ctype    en_US.UTF-8</span></span>
<span><span class="co">#&gt;  tz       UTC</span></span>
<span><span class="co">#&gt;  date     2024-10-16</span></span>
<span><span class="co">#&gt;  pandoc   3.4 @ /usr/bin/ (via rmarkdown)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">─ Packages ───────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt;  <span style="color: #555555; font-style: italic;">package             </span> <span style="color: #555555; font-style: italic;">*</span> <span style="color: #555555; font-style: italic;">version </span> <span style="color: #555555; font-style: italic;">date (UTC)</span> <span style="color: #555555; font-style: italic;">lib</span> <span style="color: #555555; font-style: italic;">source</span></span></span>
<span><span class="co">#&gt;  abind                  1.4-8    <span style="color: #555555;">2024-09-12</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  airway               * 1.25.0   <span style="color: #555555;">2024-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  backports              1.5.0    <span style="color: #555555;">2024-05-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bibtex                 0.5.1    <span style="color: #555555;">2023-01-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Biobase              * 2.65.1   <span style="color: #555555;">2024-08-28</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocGenerics         * 0.51.3   <span style="color: #555555;">2024-10-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocManager            1.30.25  <span style="color: #555555;">2024-08-28</span> <span style="color: #555555;">[2]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  BiocParallel           1.39.0   <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  BiocStyle            * 2.33.1   <span style="color: #555555;">2024-06-12</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bookdown               0.40     <span style="color: #555555;">2024-07-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  bslib                  0.8.0    <span style="color: #555555;">2024-07-29</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cachem                 1.1.0    <span style="color: #555555;">2024-05-16</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  circlize               0.4.16   <span style="color: #555555;">2024-02-20</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cli                    3.6.3    <span style="color: #555555;">2024-06-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  clue                   0.3-65   <span style="color: #555555;">2023-09-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  cluster                2.1.6    <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  codetools              0.2-20   <span style="color: #555555;">2024-03-31</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  colorspace             2.1-1    <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  colourpicker           1.3.0    <span style="color: #555555;">2023-08-21</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ComplexHeatmap         2.21.1   <span style="color: #555555;">2024-09-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  crayon                 1.5.3    <span style="color: #555555;">2024-06-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  DelayedArray           0.31.14  <span style="color: #555555;">2024-10-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  desc                   1.4.3    <span style="color: #555555;">2023-12-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  DESeq2               * 1.45.3   <span style="color: #555555;">2024-07-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  digest                 0.6.37   <span style="color: #555555;">2024-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  doParallel             1.0.17   <span style="color: #555555;">2022-02-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  dplyr                  1.1.4    <span style="color: #555555;">2023-11-17</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  DT                     0.33     <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  edgeR                * 4.3.19   <span style="color: #555555;">2024-10-11</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  evaluate               1.0.1    <span style="color: #555555;">2024-10-10</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fansi                  1.0.6    <span style="color: #555555;">2023-12-08</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fastmap                1.2.0    <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fontawesome            0.5.2    <span style="color: #555555;">2023-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  foreach                1.5.2    <span style="color: #555555;">2022-02-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  fs                     1.6.4    <span style="color: #555555;">2024-04-25</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  generics               0.1.3    <span style="color: #555555;">2022-07-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDb         * 1.41.2   <span style="color: #555555;">2024-10-02</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  GenomeInfoDbData       1.2.13   <span style="color: #555555;">2024-10-15</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  GenomicRanges        * 1.57.2   <span style="color: #555555;">2024-10-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  GetoptLong             1.0.5    <span style="color: #555555;">2020-12-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ggplot2                3.5.1    <span style="color: #555555;">2024-04-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ggrepel                0.9.6    <span style="color: #555555;">2024-09-07</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  GlobalOptions          0.1.2    <span style="color: #555555;">2020-06-10</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  glue                   1.8.0    <span style="color: #555555;">2024-09-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  gtable                 0.3.5    <span style="color: #555555;">2024-04-22</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  highr                  0.11     <span style="color: #555555;">2024-05-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  htmltools              0.5.8.1  <span style="color: #555555;">2024-04-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  htmlwidgets            1.6.4    <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  httpuv                 1.6.15   <span style="color: #555555;">2024-03-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  httr                   1.4.7    <span style="color: #555555;">2023-08-15</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  igraph                 2.0.3    <span style="color: #555555;">2024-03-13</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  IRanges              * 2.39.2   <span style="color: #555555;">2024-07-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  iSEE                 * 2.17.4   <span style="color: #555555;">2024-09-03</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  iSEEde               * 1.3.1    <span style="color: #555555;">2024-10-16</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor</span></span></span>
<span><span class="co">#&gt;  iterators              1.0.14   <span style="color: #555555;">2022-02-05</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  jquerylib              0.1.4    <span style="color: #555555;">2021-04-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  jsonlite               1.8.9    <span style="color: #555555;">2024-09-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  knitr                  1.48     <span style="color: #555555;">2024-07-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  later                  1.3.2    <span style="color: #555555;">2023-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lattice                0.22-6   <span style="color: #555555;">2024-03-20</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  lifecycle              1.0.4    <span style="color: #555555;">2023-11-07</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  limma                * 3.61.12  <span style="color: #555555;">2024-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  listviewer             4.0.0    <span style="color: #555555;">2023-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  locfit                 1.5-9.10 <span style="color: #555555;">2024-06-24</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  lubridate              1.9.3    <span style="color: #555555;">2023-09-27</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  magrittr               2.0.3    <span style="color: #555555;">2022-03-30</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Matrix                 1.7-0    <span style="color: #555555;">2024-04-26</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  MatrixGenerics       * 1.17.0   <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  matrixStats          * 1.4.1    <span style="color: #555555;">2024-09-08</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  memoise                2.0.1    <span style="color: #555555;">2021-11-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  mgcv                   1.9-1    <span style="color: #555555;">2023-12-21</span> <span style="color: #555555;">[3]</span> <span style="color: #555555;">CRAN (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  mime                   0.12     <span style="color: #555555;">2021-09-28</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  miniUI                 0.1.1.1  <span style="color: #555555;">2018-05-18</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  munsell                0.5.1    <span style="color: #555555;">2024-04-01</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  nlme                   3.1-166  <span style="color: #555555;">2024-08-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pillar                 1.9.0    <span style="color: #555555;">2023-03-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pkgconfig              2.0.3    <span style="color: #555555;">2019-09-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  pkgdown                2.1.1    <span style="color: #555555;">2024-09-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  plyr                   1.8.9    <span style="color: #555555;">2023-10-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  png                    0.1-8    <span style="color: #555555;">2022-11-29</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  promises               1.3.0    <span style="color: #555555;">2024-04-05</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  R6                     2.5.1    <span style="color: #555555;">2021-08-19</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  ragg                   1.3.3    <span style="color: #555555;">2024-09-11</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  RColorBrewer           1.1-3    <span style="color: #555555;">2022-04-03</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  Rcpp                   1.0.13   <span style="color: #555555;">2024-07-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  RefManageR           * 1.4.0    <span style="color: #555555;">2022-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rintrojs               0.3.4    <span style="color: #555555;">2024-01-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rjson                  0.2.23   <span style="color: #555555;">2024-09-16</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rlang                  1.1.4    <span style="color: #555555;">2024-06-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  rmarkdown              2.28     <span style="color: #555555;">2024-08-17</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  S4Arrays               1.5.11   <span style="color: #555555;">2024-10-14</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  S4Vectors            * 0.43.2   <span style="color: #555555;">2024-07-17</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  sass                   0.4.9    <span style="color: #555555;">2024-03-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  scales                 1.3.0    <span style="color: #555555;">2023-11-28</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  sessioninfo          * 1.2.2    <span style="color: #555555;">2021-12-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  shape                  1.4.6.1  <span style="color: #555555;">2024-02-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  shiny                  1.9.1    <span style="color: #555555;">2024-08-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  shinyAce               0.4.2    <span style="color: #555555;">2022-05-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  shinydashboard         0.7.2    <span style="color: #555555;">2021-09-30</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  shinyjs                2.1.0    <span style="color: #555555;">2021-12-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  shinyWidgets           0.8.7    <span style="color: #555555;">2024-09-23</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  SingleCellExperiment * 1.27.2   <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  SparseArray            1.5.44   <span style="color: #555555;">2024-10-06</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  statmod                1.5.0    <span style="color: #555555;">2023-01-06</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  stringi                1.8.4    <span style="color: #555555;">2024-05-06</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  stringr                1.5.1    <span style="color: #555555;">2023-11-14</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  SummarizedExperiment * 1.35.4   <span style="color: #555555;">2024-10-09</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.1)</span></span></span>
<span><span class="co">#&gt;  systemfonts            1.1.0    <span style="color: #555555;">2024-05-15</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  textshaping            0.4.0    <span style="color: #555555;">2024-05-24</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tibble                 3.2.1    <span style="color: #555555;">2023-03-20</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  tidyselect             1.2.1    <span style="color: #555555;">2024-03-11</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  timechange             0.3.0    <span style="color: #555555;">2024-01-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  UCSC.utils             1.1.0    <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  utf8                   1.2.4    <span style="color: #555555;">2023-10-22</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  vctrs                  0.6.5    <span style="color: #555555;">2023-12-01</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  vipor                  0.4.7    <span style="color: #555555;">2023-12-18</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  viridisLite            0.4.2    <span style="color: #555555;">2023-05-02</span> <span style="color: #555555;">[1]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  xfun                   0.48     <span style="color: #555555;">2024-10-03</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  xml2                   1.3.6    <span style="color: #555555;">2023-12-04</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  xtable                 1.8-4    <span style="color: #555555;">2019-04-21</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  XVector                0.45.0   <span style="color: #555555;">2024-05-01</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  yaml                   2.3.10   <span style="color: #555555;">2024-07-26</span> <span style="color: #555555;">[2]</span> <span style="color: #BB00BB; font-weight: bold;">RSPM (R 4.4.0)</span></span></span>
<span><span class="co">#&gt;  zlibbioc               1.51.1   <span style="color: #555555;">2024-06-05</span> <span style="color: #555555;">[1]</span> <span style="color: #555555;">Bioconductor 3.20 (R 4.4.0)</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [1] /__w/_temp/Library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [2] /usr/local/lib/R/site-library</span></span></span>
<span><span class="co">#&gt; <span style="color: #555555;"> [3] /usr/local/lib/R/library</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB; font-weight: bold;">──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────</span></span></span></code></pre>
</div>
<div class="section level2">
<h2 id="bibliography">Bibliography<a class="anchor" aria-label="anchor" href="#bibliography"></a>
</h2>
<p>This vignette was generated using <em><a href="https://bioconductor.org/packages/3.20/BiocStyle" class="external-link">BiocStyle</a></em>
<code>r Citep(bib[["BiocStyle"]])</code> with <em><a href="https://CRAN.R-project.org/package=knitr" class="external-link">knitr</a></em> (Xie,
2024) and <em><a href="https://CRAN.R-project.org/package=rmarkdown" class="external-link">rmarkdown</a></em>
(Allaire, Xie, Dervieux et al., 2024) running behind the scenes.</p>
<p>Citations made with <em><a href="https://CRAN.R-project.org/package=RefManageR" class="external-link">RefManageR</a></em>
(McLean, 2017).</p>
<p>
<a id="bib-allaire2024rmarkdown"></a><a href="#cite-allaire2024rmarkdown">[1]</a><cite>
J. Allaire, Y. Xie, C. Dervieux, et al. <em>rmarkdown: Dynamic Documents
for R</em>. R package version 2.28. 2024. URL:
<a href="https://github.com/rstudio/rmarkdown" class="external-link">https://github.com/rstudio/rmarkdown</a>.</cite>
</p>
<p>
<a id="bib-mclean2017refmanager"></a><a href="#cite-mclean2017refmanager">[2]</a><cite>
M. W. McLean. “RefManageR: Import and Manage BibTeX and BibLaTeX
References in R”. In: <em>The Journal of Open Source Software</em>
(2017). DOI:
<a href="https://doi.org/10.21105/joss.00338" class="external-link">10.21105/joss.00338</a>.</cite>
</p>
<p>
<a id="bib-ole2024biocstyle"></a><a href="#cite-ole2024biocstyle">[3]</a><cite>
A. Oleś. <em>BiocStyle: Standard styles for vignettes and other
Bioconductor documents</em>. R package version 2.33.1. 2024. DOI:
<a href="https://doi.org/10.18129/B9.bioc.BiocStyle" class="external-link">10.18129/B9.bioc.BiocStyle</a>.
URL:
<a href="https://bioconductor.org/packages/BiocStyle" class="external-link">https://bioconductor.org/packages/BiocStyle</a>.</cite>
</p>
<p>
<a id="bib-2024language"></a><a href="#cite-2024language">[4]</a><cite>
R Core Team. <em>R: A Language and Environment for Statistical
Computing</em>. R Foundation for Statistical Computing. Vienna, Austria,
2024. URL:
<a href="https://www.R-project.org/" class="external-link">https://www.R-project.org/</a>.</cite>
</p>
<p>
<a id="bib-ruealbrecht2024iseede"></a><a href="#cite-ruealbrecht2024iseede">[5]</a><cite>
K. Rue-Albrecht. <em>iSEEde: iSEE extension for panels related to
differential expression analysis</em>. R package version 1.3.1. 2024.
URL:
<a href="https://github.com/iSEE/iSEEde" class="external-link">https://github.com/iSEE/iSEEde</a>.</cite>
</p>
<p>
<a id="bib-wickham2011testthat"></a><a href="#cite-wickham2011testthat">[6]</a><cite>
H. Wickham. “testthat: Get Started with Testing”. In: <em>The R
Journal</em> 3 (2011), pp. 5–10. URL:
<a href="https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf" class="external-link">https://journal.r-project.org/archive/2011-1/RJournal_2011-1_Wickham.pdf</a>.</cite>
</p>
<p>
<a id="bib-wickham2021sessioninfo"></a><a href="#cite-wickham2021sessioninfo">[7]</a><cite>
H. Wickham, W. Chang, R. Flight, et al. <em>sessioninfo: R Session
Information</em>. R package version 1.2.2, <a href="https://r-lib.github.io/sessioninfo/" class="external-link uri">https://r-lib.github.io/sessioninfo/</a>. 2021. URL:
<a href="https://github.com/r-lib/sessioninfo#readme" class="external-link">https://github.com/r-lib/sessioninfo#readme</a>.</cite>
</p>
<p>
<a id="bib-xie2024knitr"></a><a href="#cite-xie2024knitr">[8]</a><cite>
Y. Xie. <em>knitr: A General-Purpose Package for Dynamic Report
Generation in R</em>. R package version 1.48. 2024. URL:
<a href="https://yihui.org/knitr/" class="external-link">https://yihui.org/knitr/</a>.</cite>
</p>
<!-- Links -->
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Kevin Rue-Albrecht, Denali Therapeutics.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
